src/structui/ui.py

```
@@ -291,7 +291,12 @@ def handler(e):
                             try: val = int(float(val))
                             except ValueError: pass
                         elif prop_type in ['number', 'float'] and val is not None and val != '':
-                            try: val = float(val)
+                            try:
+                                val_str = str(val).strip()
+                                if '.' in val_str or 'e' in val_str.lower():
+                                    val = float(val_str)
+                                else:
+                                    val = int(val_str)
                             except ValueError: pass
```

Ruff (0.15.15)

[error] Multiple statements on one line (colon)

(E701)